model/buildlogger.go

```diff
@@ -109,9 +115,77 @@ func (l *Log) SaveNew() error {
 	return errors.Wrapf(err, "problem saving new log %s", l.ID)
 }
 
-// AppendLogChunkInfo adds a new log chunk to the log's chunks array in the
+// Remove removes the log from the database. The environment should not be nil.
+func (l *Log) Remove() error {
+	if l.env == nil {
+		return errors.New("cannot remove a log with a nil environment")
+	}
+	ctx, cancel := l.env.Context()
+	defer cancel()
+
+	if l.ID == "" {
+		l.ID = l.Info.ID()
+	}
+
+	deleteResult, err := l.env.GetDB().Collection(buildloggerCollection).DeleteOne(ctx, bson.M{"_id": l.ID})
+	grip.DebugWhen(err == nil, message.Fields{
+		"ns":           model.Namespace{DB: l.env.GetConf().DatabaseName, Collection: buildloggerCollection},
+		"id":           l.ID,
+		"deleteResult": deleteResult,
+		"op":           "remove log",
+	})
+
+	return errors.Wrapf(err, "problem removing log record with _id %s", l.ID)
+}
+
+// Upload uploads a chunk of log lines to the offline blob storage bucket
+// configured for the log and updates the metadata in the database to reflect
+// the uploaded lines.
+func (l *Log) Upload(lines []LogLine) error {
+	if !l.populated {
+		return errors.New("cannot upload log lines when log unpopulated")
+	}
+	if l.env == nil {
+		return errors.New("cannot not upload log lines with a nil environment")
+	}
+	if len(lines) == 0 {
+		grip.Warning("Log.Upload called with empty LogLine slice")
```

Comment:
make this a structgured call and put some information about the log id in the message.Fields so we at least have a little bit of a record.